shared/src/main/scala/coop/rchain/shared/StreamT.scala

```diff
@@ -29,6 +29,20 @@ sealed abstract class StreamT[F[_], +A] { self =>
     case _: SNil[F]      => StreamT.empty[F, A]
   }
 
+  def filterF[AA >: A](p: A => F[Boolean])(implicit monad: Monad[F]): F[StreamT[F, AA]] =
+    self match {
+      case SCons(curr, lazyTail) =>
+        p(curr).flatMap { cond =>
```

Comment:
`p(curr).map` would now suffice because you call `pure` on every line where you return something in the `flatMap` body.